### PR TITLE
 create pid directory, if not present

### PR DESCRIPTION
**Pull request by ktsaou**

### CODE REVIEW

**Comment by ktsaou** [comment]
@paulfantom any ideas why it keeps giving me old commits as part of new PRs? If you check the files changed tab, only the last one is there.

**Comment by ktsaou** [comment]
oh! I figured it out. Due to squash...
I have to make a new branch all the time to have it show only the last commit.

**Comment by paulfantom** [comment]
you need to rebase your local repo.

I have this git alias for it:
```
uu = "!git fetch upstream; git rebase upstream/master"
```

**Comment by ktsaou** [comment]
will this break all the branches I have with pending PRs?

**Comment by paulfantom** [comment]
no, it should not.

Explanation on my git workflow:
1. Fork repo on github
2. Download fork as "origin" (`git clone paulfantom/<some_repo>`)
3. Add original repo as "upstream" remote (`git remote add upstream <some_repo>`)
4. create branches etc.
5. sync with master branch with upstream, usually before creating new branch with (`git checkout master && git fetch upstream && git rebase upstream/master && git push -f` or with my aliases `git go master && git uu ** git push -f`)
6. Develop, and push new stuff

basically what `git fetch upstream` does is synchronizes local state of remote "upstream" repo and `git rebase upstream/master` just rebases current branch onto local (but just synchronized) upstream master branch.

**Comment by paulfantom** [comment]
In short:
`git fetch upstream` only updates git internals
`git rebase upstream/master` will just only change stuff on branch on which it was executed.

**Comment by ktsaou** [comment]
ok, perfect, then `git push --force`.

**Comment by paulfantom** [comment]
as with any rebase :)
